app/controllers/payments_controller.rb

```diff
@@ -3,21 +3,21 @@ class PaymentsController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @payments_list = Transaction.where(client_id: params[:client_id]).sort_by(&:datetime).reverse!
-    @result_balance = Client.find(params[:client_id]).result_balance
+    client_id = params[:client_id]
+    @payments_list = Transaction.where(client_id: client_id).sort_by(&:datetime).reverse!
+    @result_balance = Client.find(client_id).cash
   end
 
   def create; end
 
   def add
     @transaction = Transaction.new do |transaction|
```

Comment:
If you use correct db model, you can use current_user.transaction.crate(your_params).